.github/workflows/build.yml

```diff
@@ -1,13 +1,290 @@
-name: CI
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
 
-on: [push]
+name: "Build"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
 
 jobs:
-  build:
+  tests:
+    name: "Test"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies: ["lowest", "highest"]
+        php-version:
+          - "7.2"
+          - "7.3"
+          - "7.4"
+        operating-system: ["ubuntu-latest", "windows-latest"]
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2.0.0"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@1.8.2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+          extensions: mbstring
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v1.1.2"
+        with:
+          path: "~/.composer/cache"
+          key: "php-${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}"
+          restore-keys: "php-${{ matrix.php-version }}-composer-"
+
+      - name: "Install lowest dependencies"
+        if: ${{ matrix.dependencies == 'lowest' }}
+        run: "composer update --prefer-lowest --prefer-dist --no-interaction --no-progress --no-suggest"
+
+      - name: "Install highest dependencies"
+        if: ${{ matrix.dependencies == 'highest' }}
+        run: "composer update --prefer-dist --no-interaction --no-progress --no-suggest"
+
+      - name: "Tests"
+        run: "php -dmemory_limit=-1 vendor/phpunit/phpunit/phpunit"
```

Comment:
why not `php vendor/bin/phpunit`? relying on the package path seems very fragile here

Reply:
This is content of the file on Windows:

```
#!/usr/bin/env sh

dir=$(cd "${0%[/\\]*}" > /dev/null; cd "../phpunit/phpunit" && pwd)

if [ -d /proc/cygdrive ]; then
    case $(which php) in
        $(readlink -n /proc/cygdrive)/*)
            # We are in Cygwin using Windows php, so the path must be translated
            dir=$(cygpath -m "$dir");
            ;;
    esac
fi

"${dir}/phpunit" "$@" 
```

So we really need the PHP file :(

Reply:
weeeeiiiirrrrd :| 

Ok, understandable :+1: